src/charmlibs/pathops/_container_path.py

```diff
@@ -57,34 +74,73 @@ def __init__(self, *parts: StrPathLike, container: ops.Container) -> None:
     #############################
 
     def __hash__(self) -> int:
+        """Hash on (container-name, path) for efficiency."""
         return hash((self._container.name, self._path))
 
     def __repr__(self) -> str:
+        """Return a string representation including the class, path string, and container name.
+
+        ::
+
+            container = self.unit.get_container('c')
+            repr(ContainerPath('/', 'foo', 'bar', container=container))
+            # "ContainerPath('/foo/bar', container=<ops.Container 'c'>)"
+        """
         container_repr = f'<ops.Container {self._container.name!r}>'
         return f"{type(self).__name__}('{self._path}', container={container_repr})"
 
     def __str__(self) -> str:
+        """Return the string representation of the path in the container.
+
+        This is equivalent to the string representation of the :class:`pathlib.PurePath` this
+        :class:`ContainerPath` was instantiated with.
+        """
         return self._path.__str__()
 
     def as_posix(self) -> str:
+        """Return the string representation of the path in the container."""
         return self._path.__str__()
 
     def __lt__(self, other: Self) -> bool:
+        """Compare paths only if other is a ContainerPath on the same container.
+
+        To sort collections of mixed paths, consider using :class:`str` as the key to sort
+        purely by the path, or :func:`repr` to sort primarily by path class, secondarily by path,
+        and thirdly by container name.
```

Comment:
It might be good to simplify the sentence structure here, for easier reading (and similar sentences elsewhere). How about:

```suggestion
        To sort collections of mixed paths, consider using :class:`str` as the key. This will sort
        purely by the path. Alternatively, if you use :func:`repr` as the key, the paths will be sorted
        by path class, then by path, then by container name.
```

Reply:
Per Ben's advice I'll drop this paragraph.